### ISSUE TITLE
Notebooks: When I select a remote CSV, the time range should auto-populate

### ISSUE THREAD

**Issue by juliajames12** [influxdata/ui#2556]
We'd like to update the remote CSV panel to auto-populate the time range when a remote CSV file has been selected. Without the time range auto-populating, the user would have to guess the time range.

Question: Can this be done for CSV files a user uploads as well?

![Screen Shot 2021-09-08 at 11.26.20 AM.png](https://images.zenhubusercontent.com/6058abb75f3c06c1d25467ae/b34d6f29-31bf-4723-b540-eae2dcabcd20)

**Comment by drdelambre** [comment]
i couldn't find anything in the docs more performant than this:
```
import "influxdata/influxdb/sample"
sorted = sample.data(set: "noaa")
    |> keep(columns: ["_time"])
    |> group()
    |> sort(columns: ["_time"])

sorted
    |> first(column: "_time")
    |> yield(name: "first")

sorted
    |> last(column: "_time")
    |> yield(name: "last")
```

a couple of caveats i found figuring this out:
- `__PREVIOUS_RESULT__` breaks when the preceding query has an import section. `sorted = __PREVIOUS_RESULT__ |> other` does not work. we will need to fix this.
- the time range is dependent on the csv that is referenced, and not the preview lifecycle of the notebook. But for custom csvs, that time is absolute, and for the built in data, that time is semi-relative (wiped and refreshed every 15min). So we're going to have to figure out a way to keep the info fresh, and it needs to be run before running preview as setting the time range for the following panel is a sync operation

**Comment by drdelambre** [comment]
Here's the issue for the required `__PREVIOUS_RESULT__` fix: #2609

**Comment by russorat** [comment]
```
import "experimental"
import "influxdata/influxdb/sample"

data = sample.data(set: "usgs")

min = data |> keep(columns: ["_time"]) |> group() |> min(column: "_time") |> tableFind(fn: (key) => true) |> getRecord(idx: 0)
max = data |> keep(columns: ["_time"]) |> group() |> max(column: "_time") |> tableFind(fn: (key) => true) |> getRecord(idx: 0)
  
data
  |> range(start: min._time, stop: max._time)
  |> aggregateWindow(every: 10s, fn: last, createEmpty: false)
```
